ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,782 @@
+ * */
+
+import type ArcDiagramSeriesOptions from './ArcDiagramSeriesOptions';
+import ArcDiagramPoint from './ArcDiagramPoint.js';
+import BubbleSeries from '../Bubble/BubbleSeries.js';
+import SankeyColumnComposition from '../Sankey/SankeyColumnComposition.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import type { StatesOptionsKey } from '../../Core/Series/StatesOptions';
+import type SVGAttributes from '../../Core/Renderer/SVG/SVGAttributes';
+import SVGRenderer from '../../Core/Renderer/SVG/SVGRenderer.js';
+const { prototype: { symbols } } = SVGRenderer;
+const {
+    seriesTypes: {
+        sankey: SankeySeries
+    }
+} = SeriesRegistry;
+import U from '../../Core/Utilities.js';
+const {
+    extend,
+    merge,
+    pick,
+    relativeLength
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.arcdiagram
+ *
+ * @augments Highcharts.seriesTypes.sankey
+ */
+class ArcDiagramSeries extends SankeySeries {
+
+    /**
+     *  Arc diagram series is a chart drawing style in which
+     *  the vertices of the chart are positioned along a line
+     *  on the Euclidean plane and the edges are drawn as a semicircle
+     *  in one of the two half-planes delimited by the line,
+     *  or as smooth curves formed by sequences of semicircles.
+     *
+     * @sample highcharts/demo/arc-diagram/
+     *         Arc Diagram
+     *
+     * @extends      plotOptions.sankey
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/arc-diagram
+     * @exclude      curveFactor, connectEnds, connectNulls, colorAxis, colorKey,
+     *               dataSorting, dragDrop, getExtremesFromAll, nodePadding,
+     *               centerInCategory, pointInterval, pointIntervalUnit,
+     *               pointPlacement, pointStart, relativeXValue, softThreshold,
+     *               stack, stacking, step, xAxis, yAxis
+     * @optionparent plotOptions.arcdiagram
+     */
+    public static defaultOptions: ArcDiagramSeriesOptions = merge(SankeySeries.defaultOptions, {
+
+        /**
+         * The option to center links rather than position them
+         * one after another
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        centeredLinks: false,
+
+        /**
+         * The radius of the link arc. If not set,
+         * it is calculated automatically.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.linkRadius
+         */
+
+        /**
+         * The offset of an arc diagram nodes column in relation
+         * to the plotArea. The offset equal to 50% places nodes
+         * in the center of a chart.
+         *
+         * @type    {string}
+         * @since next
+         * @default undefined
+         * @product highcharts
```

Comment:
It looks to me that the default behaviour is like setting this to `100%`? We can spell out that in the options (in JavaScript), then we don't need to handle `undefined`, and it makes the typing simpler.